### PR TITLE
Update our provider to match dnsimple-ruby 4.5.0

### PR DESCRIPTION
**Pull request by martinisoft**
This changeset resolves the breakage that occured in the 4.5.0 release which should resolve https://github.com/dnsimple/dnsimple-ruby/issues/169

### CODE REVIEW

File: libraries/provider_dnsimple_record.rb [hunk 1]
```diff
@@ -68,9 +68,9 @@ def load_current_resource
 
       def create_record
         converge_by("create record #{new_resource.record_name} for domain #{new_resource.domain}") do
-          dnsimple_client.zones.create_record(
-            dnsimple_client_account_id, new_resource.domain, **record_options
-          )
+          dnsimple_client.zones.create_zone_record(dnsimple_client_account_id,
+                                                   new_resource.domain,
+                                                   **record_options)
```
Comment (onlyhavecans) [inline libraries/provider_dnsimple_record.rb]: We should maintain parens on it's own line when using multiline methods for readability
```suggestion
                                                   **record_options
                                                   )
```

File: libraries/provider_dnsimple_record.rb [hunk 2]
```diff
@@ -79,9 +79,9 @@ def create_record
 
       def delete_record
         converge_by("delete record #{@new_resource.record_name} from domain #{@new_resource.domain}") do
-          dnsimple_client.zones.delete_record(dnsimple_client_account_id,
-                                              @current_resource.domain,
-                                              existing_record_id)
+          dnsimple_client.zones.delete_zone_record(dnsimple_client_account_id,
+                                                   @current_resource.domain,
+                                                   existing_record_id)
```
Comment (onlyhavecans) [inline libraries/provider_dnsimple_record.rb]: I suggest parens on own line
```suggestion
                                                   existing_record_id
                                                   )
```

File: libraries/provider_dnsimple_record.rb [hunk 3]
```diff
@@ -92,10 +92,10 @@ def delete_record
...
         return unless changed_record?
         converge_by("update record #{new_resource.record_name} for domain #{new_resource.domain}") do
-          dnsimple_client.zones.update_record(dnsimple_client_account_id,
-                                              new_resource.domain,
-                                              existing_record_id,
-                                              **record_options)
+          dnsimple_client.zones.update_zone_record(dnsimple_client_account_id,
+                                                   new_resource.domain,
+                                                   existing_record_id,
+                                                   **record_options)
```
Comment (onlyhavecans) [inline libraries/provider_dnsimple_record.rb]: same as above
```suggestion
                                                   **record_options
                                                   )
```

File: spec/libraries/provider_dnsimple_record_spec.rb [hunk 4]
```diff
@@ -29,7 +29,7 @@
     let(:response) { instance_double(Dnsimple::Response, data: data) }
     let(:data) { instance_double(Dnsimple::Struct::Whoami, account: account) }
     let(:account) { instance_double(Dnsimple::Struct::Account, id: 1) }
-    let(:zones) { instance_double(Dnsimple::Client::ZonesService, all_records: zone_records, create_record: zone_record) }
+    let(:zones) { instance_double(Dnsimple::Client::ZonesService, list_zone_records: zone_records, create_zone_record: zone_record) }
```
Comment (weppos) [inline spec/libraries/provider_dnsimple_record_spec.rb]:
```suggestion
    let(:zones) { instance_double(Dnsimple::Client::ZonesService, all_zone_records: zone_records, create_zone_record: zone_record) }
```

File: test/integration/create_record/libraries/dnsimple_zone.rb [hunk 5]
```diff
@@ -40,7 +40,7 @@ def response
 
     filter = {}
     filter = { type: @type } if @type
-    client.zones.records(account_id, @zone, filter: filter).data.detect do |record|
+    client.zones.list_zone_records(account_id, @zone, filter: filter).data.detect do |record|
```
Comment (weppos) [inline test/integration/create_record/libraries/dnsimple_zone.rb]:
```suggestion
    client.zones.all_zone_records(account_id, @zone, filter: filter).data.detect do |record|
```

File: test/integration/update_record/libraries/dnsimple_zone.rb [hunk 6]
```diff
@@ -40,7 +40,7 @@ def response
 
     filter = {}
     filter = { type: @type } if @type
-    client.zones.records(account_id, @zone, filter: filter).data.detect do |record|
+    client.zones.list_zone_records(account_id, @zone, filter: filter).data.detect do |record|
```
Comment (weppos) [inline test/integration/update_record/libraries/dnsimple_zone.rb]:
```suggestion
    client.zones.all_zone_records(account_id, @zone, filter: filter).data.detect do |record|
```

File: spec/libraries/provider_dnsimple_record_spec.rb [hunk 7]
```diff
@@ -171,7 +171,7 @@
     let(:response) { instance_double(Dnsimple::Response, data: data) }
     let(:data) { instance_double(Dnsimple::Struct::Whoami, account: account) }
     let(:account) { instance_double(Dnsimple::Struct::Account, id: 1) }
-    let(:zones) { instance_double(Dnsimple::Client::ZonesService, all_records: zone_records, update_record: zone_record) }
+    let(:zones) { instance_double(Dnsimple::Client::ZonesService, list_zone_records: zone_records, update_zone_record: zone_record) }
```
Comment (weppos) [inline spec/libraries/provider_dnsimple_record_spec.rb]:
```suggestion
    let(:zones) { instance_double(Dnsimple::Client::ZonesService, all_zone_records: zone_records, update_zone_record: zone_record) }
```

File: spec/libraries/provider_dnsimple_record_spec.rb [hunk 8]
```diff
@@ -90,15 +90,15 @@
...
+      allow(zones).to receive(:delete_zone_record)
     end
 
     let(:client) { instance_double(Dnsimple::Client, identity: identity, zones: zones) }
     let(:identity) { instance_double(Dnsimple::Client::Identity, whoami: response) }
     let(:response) { instance_double(Dnsimple::Response, data: data) }
     let(:data) { instance_double(Dnsimple::Struct::Whoami, account: account) }
     let(:account) { instance_double(Dnsimple::Struct::Account, id: 1) }
-    let(:zones) { instance_double(Dnsimple::Client::ZonesService, all_records: zone_records) }
+    let(:zones) { instance_double(Dnsimple::Client::ZonesService, list_zone_records: zone_records) }
```
Comment (weppos) [inline spec/libraries/provider_dnsimple_record_spec.rb]:
```suggestion
    let(:zones) { instance_double(Dnsimple::Client::ZonesService, all_zone_records: zone_records) }
```

**Comment by martinisoft** [comment]
@thekendalmiller Let us know how my latest updates work for you now. Did not know about the difference in methods so that should be corrected now. If all is well, I will cut a new patch release since this is also affecting us in production (yes, we dogfood our own cookbook)

**Comment by thekendalmiller** [comment]
@martinisoft Thanks. It works for me now

**Pull request merged**